bin/control.bat

```diff
@@ -102,92 +81,82 @@ if /i "%SCRIPTS_HOME%\" == "%~dp0" goto setProgName
 for /f "usebackq tokens=*" %%i in (`!RANDOM_NUMBER_COMMAND!`) do set RANDOM_NUMBER=%%i
 
 set RESTART_SUCCESS_FILE="%IGNITE_HOME%\work\ignite_success_%RANDOM_NUMBER%"
 set RESTART_SUCCESS_OPT=-DIGNITE_SUCCESS_FILE=%RESTART_SUCCESS_FILE%
 
-::
-:: JVM options. See http://java.sun.com/javase/technologies/hotspot/vmoptions.jsp for more details.
-::
-:: ADD YOUR/CHANGE ADDITIONAL OPTIONS HERE
-::
-"%JAVA_HOME%\bin\java.exe" -version 2>&1 | findstr "1\.[7]\." > nul
-if %ERRORLEVEL% equ 0 (
-    if "%CONTROL_JVM_OPTS%" == "" set CONTROL_JVM_OPTS=-Xms256m -Xmx1g
-) else (
-    if "%CONTROL_JVM_OPTS%" == "" set CONTROL_JVM_OPTS=-Xms256m -Xmx1g
-)
+:: Determine Java version function
+:determineJavaVersion
+for /f "tokens=1,2 delims=." %%a in ("%JAVA_VER_STR%.x") do set JAVA_VERSION=%%a
+
+:: Set JVM options with dynamic version check
+call :getJavaSpecificOpts %JAVA_VERSION% "%CONTROL_JVM_OPTS%"
+
+:: Enable assertions if set.
+if "%ENABLE_ASSERTIONS%" == "1" set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -ea
+
+:: Set main class to start service (grid node by default).
+if "%MAIN_CLASS%" == "" set MAIN_CLASS=org.apache.ignite.internal.commandline.CommandHandler
 
-::
 :: Uncomment to enable experimental commands [--wal]
-::
 :: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -DIGNITE_ENABLE_EXPERIMENTAL_COMMAND=true
 
-::
 :: Uncomment the following GC settings if you see spikes in your throughput due to Garbage Collection.
-::
 :: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -XX:+UseG1GC
 
-::
 :: Uncomment if you get StackOverflowError.
 :: On 64 bit systems this value can be larger, e.g. -Xss16m
-::
 :: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -Xss4m
 
-::
 :: Uncomment to set preference to IPv4 stack.
-::
 :: set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -Djava.net.preferIPv4Stack=true
 
-::
-:: Assertions are disabled by default since version 3.5.
-:: If you want to enable them - set 'ENABLE_ASSERTIONS' flag to '1'.
-::
-set ENABLE_ASSERTIONS=1
-
-::
-:: Set '-ea' options if assertions are enabled.
-::
-if %ENABLE_ASSERTIONS% == 1 set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% -ea
-
-:run_java
-
-::
-:: Set main class to start service (grid node by default).
-::
-
-if "%MAIN_CLASS%" == "" set MAIN_CLASS=org.apache.ignite.internal.commandline.CommandHandler
-
-::
-:: Remote debugging (JPDA).
-:: Uncomment and change if remote debugging is required.
-:: set CONTROL_JVM_OPTS=-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=8787 %CONTROL_JVM_OPTS%
-::
-
-::
 :: Final CONTROL_JVM_OPTS for Java 9+ compatibility
-::
-call "%SCRIPTS_HOME%\include\jvmdefaults.bat" %MAJOR_JAVA_VER% "%CONTROL_JVM_OPTS%" CONTROL_JVM_OPTS
+if %JAVA_VERSION% GEQ 9 (
+    set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% --illegal-access=warn
+    set CONTROL_JVM_OPTS=%CONTROL_JVM_OPTS% --add-opens=java.base/java.nio=ALL-UNNAMED
+)
+
+if %java_version% GEQ 17 (
+    rem For JDK 17 and above, add necessary opens
+    set value=
```

Comment:
The getJavaSpecificOpts from jvmdefaults.bat should be used to set JVM options here, similar to control.sh script.